zlibWrapper/zstd_zlibwrapper.c

```diff
@@ -500,6 +531,7 @@ ZEXTERN int ZEXPORT z_deflatePending OF((z_streamp strm,
         return deflatePending(strm, pending, bits);
     FINISH_WITH_ERR("deflatePending is not supported!");
```

Comment:
Well, the precise error code to use is obviously situation dependent.

Currently, it seems you selected `Z_MEM_ERROR` for all errors within `FINISH_WITH_ERR` macros.

I suspect `Z_ERRNO` to be the "default error" when not specific,
`Z_MEM_ERROR` more likely related to memory management issues,
`Z_DATA_ERROR` more likely related to corruption issues,
and `Z_VERSION_ERROR` potentially related to unauthorized header fields or values.
No idea for `Z_BUF_ERROR` or `Z_STREAM_ERROR`.

In any case, the precise error number is at this stage less important than removing the `fprintf` .